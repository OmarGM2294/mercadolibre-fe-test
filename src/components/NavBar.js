import React from 'react'
import Link from 'next/link'
import Image from 'next/image'
import styles from '../../styles/components/NavBar.module.scss'

const NavBar = () => {
  return (
    <header className={styles.nav}>
      <nav className={styles.container}>
        <Link href="/">
          <a className={styles.link}>
            <Image
              className={styles.logo}
              src="https://http2.mlstatic.com/frontend-assets/ui-navigation/5.14.3/mercadolibre/logo__small.png"
              alt="logo"
              width={44}
              height={32}
            />
          </a>
        </Link>
        <form
          className={styles.search}
          action="/items"
          method="get">
          <input
            type="text"
            autoFocus
            name="search"
            aria-label="Busca lo que necesitas"
            placeholder="Nunca dejes de buscar" />
          <button
            className={styles.btn}
            aria-label="Buscar"
            type="submit">
            <Image
              src="/imgs/search.svg"
              alt="Buscar"
              height={16}
              width={16}
            />
          </button>
        </form>
      </nav>
    </header>
  )
}

export default NavBar