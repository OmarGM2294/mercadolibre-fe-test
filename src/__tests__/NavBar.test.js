import React from 'react'

import { create } from 'react-test-renderer'

import NavBar from '../components/NavBar'

describe('NavBar', () => {
  it('Should match snapshot (it renders correctly)', () => {
    const component = create(<NavBar />)

    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})