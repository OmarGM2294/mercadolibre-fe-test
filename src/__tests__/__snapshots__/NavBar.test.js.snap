// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Should match snapshot (it renders correctly) 1`] = `
<header>
  <nav>
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "inline-block",
            "margin": 0,
            "maxWidth": "100%",
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "maxWidth": "100%",
            }
          }
        >
          <img
            alt=""
            aria-hidden={true}
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNDQiIGhlaWdodD0iMzIiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
            style={
              Object {
                "border": "none",
                "display": "block",
                "margin": 0,
                "maxWidth": "100%",
                "padding": 0,
              }
            }
          />
        </div>
        <noscript>
          <img
            alt="logo"
            decoding="async"
            src="/imgs/logo.png"
            srcSet="/_next/image?url=%2Fimgs%2Flogo.png&w=48&q=100 1x, /_next/image?url=%2Fimgs%2Flogo.png&w=96&q=100 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
        <img
          alt="logo"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </div>
    </a>
    <form
      action="/items"
      method="get"
    >
      <input
        aria-label="Busca lo que necesitas"
        autoFocus={true}
        name="search"
        placeholder="Nunca dejes de buscar"
        type="text"
      />
      <button
        aria-label="Buscar"
        type="submit"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYiIGhlaWdodD0iMTYiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <noscript>
            <img
              alt="Buscar"
              decoding="async"
              src="/imgs/search.svg"
              srcSet="/_next/image?url=%2Fimgs%2Fsearch.svg&w=16&q=75 1x, /_next/image?url=%2Fimgs%2Fsearch.svg&w=32&q=75 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
          <img
            alt="Buscar"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </div>
      </button>
    </form>
  </nav>
</header>
`;
